src/main/kotlin/com/zopa/ktor/opentracing/ThreadContextElementScopeManager.kt

```diff
@@ -3,26 +3,27 @@ package com.zopa.ktor.opentracing
 import io.opentracing.Scope
 import io.opentracing.ScopeManager
 import io.opentracing.Span
-import io.opentracing.noop.NoopScopeManager
 import java.util.Stack
 
 
 internal val threadLocalSpanStack = ThreadLocal<Stack<Span>>()
 
 class ThreadContextElementScopeManager: ScopeManager {
     override fun activate(span: Span?): Scope {
-        val spanStack = threadLocalSpanStack.get()
+        var spanStack = threadLocalSpanStack.get()
+
         if (spanStack == null) {
-            log.error { "spanStack is null" }
-            return NoopScopeManager.NoopScope.INSTANCE
+            spanStack = Stack<Span>()
+            threadLocalSpanStack.set(spanStack)
```

Comment:
Currently, if you try to activate a span without having any existing spans ( ie a null threadLocalSpanStack ) we return a Noop scope and do not store the span in the span stack.

Reply:
Might be worth adding a quick unit test to cover this functionality to ensure that the bug is indeed fixed?

Also could add an info log to indicate when this code path is being run through.

Reply:
Yes good idea , added a test here https://github.com/zopaUK/ktor-opentracing/pull/17/commits/a74c16a45793663171213cb5c48042323fc8e2f9 and the info log here https://github.com/zopaUK/ktor-opentracing/pull/17/commits/b808d224f381dd7742db3f8f059099bb0b95ac4a